brokers/che-plugin-broker/broker.go

```diff
@@ -36,29 +36,45 @@ const depFileBothLocationAndURLError = "Plugin dependency '%s:%s' contains both
 const depFileNoLocationURLError = "Plugin dependency '%s:%s' contains neither 'location' nor 'url' field"
 
 type PluginLinkType int
+
 const (
 	Archive PluginLinkType = iota + 1
 	Yaml
 )
 
 // ChePluginBroker is used to process Che plugins
-type ChePluginBroker struct {
+type ChePluginBroker interface {
+	Start([]model.PluginMeta)
+	PushEvents(tun *jsonrpc.Tunnel)
+	ProcessPlugin(meta model.PluginMeta) error
+}
+
+type chePluginBrokerImpl struct {
```

Comment:
So it seems I'll have to rework my PR a bit more than expected, since I have made the `storage` field public in order to be able to retrieve the result from the broker after using `ProcessPlugin`.

Reply:
Oh, or I would would now be able create a broker with a common external storage in which the results of all the brokers would be accumulated. Let me wait for you to merge, and I'll accomodate with your last changes.

Reply:
It is kinda violation of encapsulation. If one uses field in tests it is bad but acceptable I suppose, but using it from other packages makes code too coupled. 
After this PR you can create brokers with provided storage as an argument to a method.
And frankly speaking you can use unified broker to process all the plugins and only then create kubernetes objects. It can improve quality of the code by decoupling different areas of responsibility.

Reply:
OK, I'll see that after you have merged. Thanks for the details.